canonical-web-and-design/maas-ui
# refactor: connect Header and Footer components to the store

## Done

- Connect `Header` and `Footer` components to the store
- Move analytics `useEffect`s to their own hooks

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that you can still log in and log out as before
- Check that all the links work as before (i.e. they go to the right place, highlights in sub-urls)
- Create a new user and log in as them
- Check that you get redirected to the user intro
- Open `src/app/App.tsx` and edit line 35 to:
  ``` ts
  const debug = false;
  ```
- Enable analytics in settings if it isn't already
- Check that the "Give feedback" link shows up in the footer 

## Fixes

Fixes canonical-web-and-design/app-tribe#1042
